src/Http/Middleware/ProxyAddress.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Http\Middleware;
+
+use Flarum\Http\Exception\ProxyNotAllowedException;
+use Illuminate\Support\Arr;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class ProxyAddress implements Middleware
+{
+    /**
+     * @var bool
+     */
+    protected $enabled;
+
+    /**
+     * @var array
+     */
+    protected $allowedAddresses;
+
+    /**
+     * @param bool $enabled
+     * @param array $allowedAddresses
+     */
+    public function __construct($enabled, $allowedAddresses)
+    {
+        $this->enabled = $enabled;
+        $this->allowedAddresses = $allowedAddresses;
+    }
+
+    private function wildcardMatch(string $ipAddress): bool
+    {
+        foreach ($this->allowedAddresses as $allowedAddress) {
+            if (fnmatch($allowedAddress, $ipAddress)) {
```

Comment:
My only concern with `fnmatch` is that it's unavailable for Windows systems. I know that Windows is a very, very tiny part of our community but I think it's still something we would want to be able to support.

Reply:
We already use fnmatch for csrf checks though.

Reply:
Also, as per https://www.php.net/manual/en/function.fnmatch.php, it should be available?

Reply:
I did develop and test this function on Windows with XAMPP, so it definitely works.

Reply:
AH my fault, I read the warning incorrectly, I thought it said that it was available to non-POSIX compliant systems except Windows, reading it again I realize I had that backwards.